src/java.base/share/classes/java/util/Formatter.java

```diff
@@ -3012,19 +3014,19 @@ private void printCharacter(Object arg, Locale l) throws IOException {
             if (arg instanceof Character) {
                 s = ((Character)arg).toString();
             } else if (arg instanceof Byte) {
-                byte i = ((Byte)arg).byteValue();
+                byte i = (Byte) arg;
```

Comment:
Can the pattern matching for instanceof be used here to remove explicit casts.  (Supported in JDK 16) 
```String s = null;
            if (arg instanceof Character c) {
                s = c.toString();
            } else if (arg instanceof Byte i) {
                if (Character.isValidCodePoint(i))
                    s = new String(Character.toChars(i));
                else
                    throw new IllegalFormatCodePointException(i);
            } else if (arg instanceof Short i) {
                if (Character.isValidCodePoint(i))
                    s = new String(Character.toChars(i));
                else
                    throw new IllegalFormatCodePointException(i);
            } ```
etc..

Reply:
I did think about it, but it seemed to stray a bit too far from the intent of this enhancement.

Reply:
I only looked at it because of the updates to use switch expressions...
ok, either way.

Reply:
I had reason to much around with the switch expressions, since `Conversion.isValid` was inefficient (for startup) and subtly wrong (accepted `'t'`).

Getting rid of explicit unboxing - `.byteValue()` - is just a syntactic improvement, so I indulged in a few places.

Using instanceof pattern matching OTOH changes bytecode shape a fair bit by introducing locals and changing the execution order. The suggestions you made above also mean we'd unbox twice. Probably inconsequential, but I'm wary of such changes when I don't have benchmarks to assert they are performance neutral.

Reply:
Makes sense, good to know... slick language features *may* come at a cost.